File: cpp/examples/basic_usage.cpp
#include <iostream>
#include <vector>
#include <cassert>
#include "msg/msg.hpp"

int main() {
    // Create message
    Msg original;
    original.body = "Hello from C++!";
    original.fromId = "cpp_example";
    original.id = "example_001";
    original.toIds = {"user1", "user2"};
    original.type = "example";

    std::cout << "Original message body: " << original.body << std::endl;

    // Serialize
    auto bytes = original.encode();
    std::cout << "Serialized size: " << bytes.size() << " bytes" << std::endl;

    // Deserialize
    Msg decoded;
    decoded.decode(bytes);

    // Verify
    assert(decoded.body == original.body);
    assert(decoded.fromId == original.fromId);
    assert(decoded.id == original.id);
    assert(decoded.toIds == original.toIds);
    assert(decoded.type == original.type);

    std::cout << "✅ C++ serialization test passed!" << std::endl;
    return 0;
}

File: rust/examples/basic_usage.rs
fn main() -> Result<(), Box<dyn std::error::Error>> {
    use trx::trx::msg::Msg;

    let msg = Msg::new(
        "example_001",
        "rust_example",
        vec!["user1".to_string(), "user2".to_string()],
        "example",
        b"Hello from Rust!".to_vec(),
    );

    let json = serde_json::to_string(&msg)?;
    let decoded: Msg = serde_json::from_str(&json)?;

    assert_eq!(msg, decoded);
    println!("Serialized trx msg: {json}");
    println!("Decoded body bytes: {:?}", decoded.body);
    println!("✅ Rust trx serialization test passed!");

    Ok(())
}
